### PR TITLE
Implement logging

### PR DESCRIPTION
**Pull request by yfukai**
Initial attempt for logging, associated with #24. 
- logger level is set to be each `__name__` of the files.
- removed `import logging` to avoid confusion b/w `logging` and `logger`
- comment: in `__main__.py`, the logger name is hard-coded in the script, and affected by the package name (see #31 ).

### CODE REVIEW

**Comment by yfukai** [comment]
Maybe I should add some actual logs... Will work a bit later.

**Comment by yfukai** [comment]
Thank you for your opinion! Sure, I will switch to `import logging` type of coding.

**Comment by yfukai** [comment]
What is a good way for `class`-level logging? I'll do some survey...

**Comment by yfukai** [comment]
I'm not sure what is a good way for `class`-level logging for now, but just added `basicpy.basicpy.BaSiC.(object-id)` level logger. Also tried to add some logs for calculation. Opinions are welcome!

File: src/pybasic/pybasic.py [hunk 1]
```diff
@@ -125,6 +130,7 @@ class BaSiC(BaseModel):
     _reweight_score: float = PrivateAttr(None)
     _flatfield: np.ndarray = PrivateAttr(None)
     _darkfield: np.ndarray = PrivateAttr(None)
+    _logger: logging.Logger = PrivateAttr(None)
```
Comment (tdmorello) [inline src/pybasic/pybasic.py:133]: Pydantic won't dump `_logger`, right?

Comment (tdmorello) [reply src/pybasic/pybasic.py:133]: Did you figure anything out for this?

Comment (yfukai) [reply src/pybasic/pybasic.py]: I tried with the `pydantic` [test code](https://pydantic-docs.helpmanual.io/usage/models/#private-model-attributes) and it seems the members annotated by `PrivateAttr` does not appear either of `model.dict()` or `model.json()`!

**Review by Nicholas-Schaub — CHANGES_REQUESTED** [review]
Just a couple minor comments. I think it looks good.

File: src/pybasic/pybasic.py (hunk 1 above)
Comment (Nicholas-Schaub) [reply src/pybasic/pybasic.py]: I still don't understand the need for additional complexity here. Creating a unique logger for each model seems like an unnecessary complication. Why not use the file level logger here?

Comment (yfukai) [reply src/pybasic/pybasic.py]: I thought it may be suitable in cases where we have multiple BaSiC objects and want to track the logs independently.

Comment (yfukai) [reply src/pybasic/pybasic.py]: > I don't think we need to have logger be part of the pydantic class. I think the code can just use the file-level logger.
Not sure why this was marked as resolved.

Hi @Nicholas-Schaub, sorry, it seems I missed your comment yesterday and marked it as resolved as I answered Tim's first comment. Assuming most of the users don't use the multiple objects simultaneously, maybe we can output the log in `basicpy.BaSiC` level.

File: src/pybasic/__main__.py [hunk 2]
```diff
@@ -122,8 +121,10 @@ def snake_to_camel(string: str):
     )
 
     args = parser.parse_args()
+    # FIXME fix name after deciding on package directory name
+    logger = logging.getLogger("pybasic")
     if args.verbose:
-        logging.getLogger("basicpy").setLevel(logging.DEBUG)
+        logger.setLevel(logging.DEBUG)
     else:
         logger.setLevel(logging.INFO)
```
[the remark this answers is not in the document]
Comment (yfukai) [reply src/pybasic/__main__.py]: Thanks, I didn't notice this conflict. Then I think the best logic is:
- if `verpose` is set, make the level `DEBUG`
- else, follow the environmental variable

How does this sounds? @Nicholas-Schaub

[the remark this answers is not in the document]
Comment (yfukai) [reply src/pybasic/__main__.py]: Fixed in [032031c](https://github.com/peng-lab/BaSiCPy/pull/44/commits/032031cdd2b07862cb768cf99fe39a659bb1281a)!

**Comment by yfukai** [comment]
Also, I resolved the TODO comment in `dct2d_tools`.

File: src/pybasic/pybasic.py (hunk 1 above)
Comment (yfukai) [reply src/pybasic/pybasic.py]: I just switched to the file-level logger.

**Comment by yfukai** [comment]
Hi @Nicholas-Schaub, I changed the logger to file level, could you have a quick look to see if it looks fine?

**Review by Nicholas-Schaub — APPROVED** [review]

**Pull request merged**